crates/core/component/governance/src/component/view.rs

```diff
@@ -875,8 +879,30 @@ pub trait StateWriteExt: StateWrite {
                 tracing::info!(target_height = height, "upgrade plan proposal passed");
                 self.signal_upgrade(*height).await?;
             }
+            ProposalPayload::FreezeIbcClient { client_id } => {
+                let client_id = &ClientId::from_str(client_id)
+                    .map_err(|e| tonic::Status::aborted(format!("invalid client id: {e}")))?;
+                let current_height = self
+                    .get_block_height()
+                    .await
+                    .expect("block height should be set");
+                let client_state = self.get_client_state(client_id).await?;
+                let revision_number = self.get_revision_number().await?;
+
+                let freeze_height = IbcHeight::new(revision_number, current_height)?;
```

Comment:
I think this height is wrong, it should be the height at which the client was frozen, in terms of that client's height (not the host chain's height). So we should be pulling the height from the client_state here instead.

Reply:
Ah yes, makes sense